client/src/main/java/io/pravega/client/connection/impl/RawClient.java

```diff
@@ -87,16 +88,22 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
         }
     }
 
-    public RawClient(PravegaNodeUri uri, ConnectionPool connectonPool) {
+    public RawClient(PravegaNodeUri uri, ConnectionPool connectionPool) {
         this.segmentId = null;
-        this.connection = connectonPool.getClientConnection(flow, uri, responseProcessor);
+        this.connection = connectionPool.getClientConnection(flow, uri, responseProcessor)
+        .exceptionally(e -> {
+            throw new CompletionException(new ConnectionFailedException("Failed to obtain client connection to segment " + segmentId.getScopedName()));
```

Comment:
we should al least keep a reference to `e` as "cause" of the ConnectionFailedException
otherwise we are simply "masking" any error with ConnectionFailedException

Not sure how we are dealing with special exceptions like InterruptedException

Reply:
It's not a blocking call, so we shouldn't have to worry about Interrupted.

Reply:
improved it.